diagnostics.go

```diff
@@ -0,0 +1,323 @@
+// Copyright 2017 Pilosa Corp.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilosa
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+// Default version check URL.
+const (
+	defaultVersionCheckURL = "https://diagnostics.pilosa.com/v0/version"
+)
+
+type versionResponse struct {
+	Version string `json:"version"`
+	Message string `json:"message"`
+}
+
+// DiagnosticsCollector represents a collector/sender of diagnostics data
+type DiagnosticsCollector struct {
+	mu          sync.Mutex
+	host        string
+	VersionURL  string
+	version     string
+	lastVersion string
+	startTime   int64
+	start       time.Time
+
+	metrics map[string]interface{}
+
+	client *http.Client
+
+	logOutput io.Writer
+
+	server *Server
+}
+
+// New returns a pointer to a new DiagnosticsCollector Client given an addr in the format "hostname:port".
```

Comment:
I think this comment should start "NewDiagnosticsCollector..."

Reply:
Yeah, that is not right. Reworded.